lfsapi/auth.go

```diff
@@ -57,18 +59,24 @@ func (c *Client) DoAPIRequestWithAuth(remote string, req *http.Request) (*http.R
 	return c.DoWithAuth(remote, access, req)
 }
 
-func (c *Client) doWithAuth(remote string, access creds.Access, req *http.Request, via []*http.Request) (*http.Response, error) {
+func (c *Client) doWithAuth(remote string, count *int, access creds.Access, req *http.Request, via []*http.Request) (*http.Response, error) {
+	if *count == 3 {
```

Comment:
I might suggest putting this value of `3` into a `const` with a name like `defaultMaxAuthAttempts` or something along those lines.

We might eventually want to make this configuration with a setting like `lfs.auth.retries` or `lfs.auth.maxRetries` (akin to the existing `lfs.ssh.retries` setting, for instance).

Reply:
Good idea.